### ISSUE TITLE
Expose PutField from PointData as public

### ISSUE THREAD

**Issue by lmandras-iura** [influxdata/influxdb-client-csharp#276]
__Proposal:__
Make the method <code>PointData PutField(string name, object value)</code> public or create an overload of the <code>Field(...)</code> methods that accepts an object as well.

__Current behavior:__
There is not overload like: <code>Field(string name, object value);</code>

__Desired behavior:__
Have the overload or expose PutField which is what the other Field(...) methods are wrapping.

__Use case:__
There are times in practice where you want to add fields depending on conditions and data at runtime and the value parameter type is unknown at compile-time, basically <code>object</code>.

**Comment by bednar** [comment]
Hi @lmandras-iura,

thanks for using our client.

> There are times in practice where you want to add fields depending on conditions and data at runtime and the value parameter type is unknown at compile-time, basically object.

What happen if the `object` type is not supported? 

What do you thing to use something like:

```csharp
var pointData = PointData.Measurement("my-measurement");
switch (value)
{
    case string s:
        pointData.Field("my-field", s);
        break;
    case bool b:
        pointData.Field("my-field", b);
        break;
    default:
      // Developer knows how to handle unsupported types:
        pointData.Field("my-field", value.ToString());
        break;
}
```

?

Regards

**Comment by lmandras-iura** [comment]
Hi @bednar,

What you are proposing is not going to work for my use case, plus is not Open-Closed Principle compliant.

Here is the sample code that I have:

```csharp
private PointData AddFields(PointData point, object readingData, string rootName)
        {
            var props = readingData.GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance);
            foreach (var propertyInfo in props)
            {
                var value = propertyInfo.GetValue(readingData, null);
                if (value is null)
                {
                    continue;
                }
                if (propertyInfo.IsPrimitive() || propertyInfo.PropertyType.IsNullable())
                {
                    point = this.InvokePutFiled(point, $"{rootName}{propertyInfo.Name}", value);
                }
                else
                {
                    point = this.AddFields(point, value, propertyInfo.Name);
                }
            }
            return point;
        }
```

The reading data object is a custom POCO with properties of different types(primitive and complex).

The value that I am getting is different every time (string, double etc) but I can't know it at compile time.

The <code>InvokePutField</code> is using reflection to hack the private method but basically this describes the need for that overload that is general.

**Comment by bednar** [comment]
Thanks for the explanation of your usage.

I am only concerned when someone uses the `AddField` with complex type. In your code snippet you are checking the property type to primitive but this is also a valid code: `point.PutField(propertyInfo.Name, value);`.

**Comment by lmandras-iura** [comment]
There is the option for the API to reject (throw exception) not supported types by influx like complex objects. But having the overload is very useful for scenarios like this and will avoid having each client to that switch and even worse with reflection that switch will just look uglier and more complicated as you have to get the type at runtime.

**Comment by bednar** [comment]
The issue is fixed in [4.0.0-rc3](https://github.com/influxdata/influxdb-client-csharp/milestone/31) milestone. You can use latest dev version by:

```console
dotnet add package InfluxDB.Client --version 4.0.0-dev.5607
```

For more info about upcoming `v4.0.0` see: https://github.com/influxdata/influxdb-client-csharp/blob/master/CHANGELOG.md
